### ISSUE TITLE
Log span

### ISSUE THREAD

**Issue by jlaamanen** [jlaamanen/logita#1]
Would be nice to have a log span function, along with the current "normal" log functions.

* Creating a "log span" would start a timer
* Multiple "split times" could be logged (along with any given text), e.g. after each asynchronous request is complete
* When the span ends, the "end time" gets logged (similar to "split time")
* This time, only `string` message should be supported? (doesn't make a lot of sense to show objects etc. in such split time messages)

Maybe the best way to do this in logger object API is to add a `span` function beside the log level functions. This would make log levels named `span` prohibited, but maybe that's OK...

**API usage example draft**

```typescript
import log from "./log";

// Log level must be given as a parameter now
const requestSpan = log.span("debug", "Many async actions");
// => [Mon, 21 May 2018 19:31:09 GMT] /app/src/index.ts:3:24 [DEBUG] "Many async actions" started

doSomethingThatTakesTenSeconds().then(() => {
  requestSpan.splitTime("First split time");
  // => [Mon, 21 May 2018 19:31:19 GMT] /app/src/index.ts:5:14 [DEBUG] "Many async actions": First split time 10.0s (+10.0s)
  return doSomethingThatTakesFiveSeconds();
}).then(() => {
  requestSpan.splitTime("Second split time");
  // => [Mon, 21 May 2018 19:31:24 GMT] /app/src/index.ts:8:14 [DEBUG] "Many async actions": Second split time 15.0s (+5.0s)
  return doSomethingThatTakesThreeSeconds();
}).then(() => {
  requestSpan.success();
  // [Mon, 21 May 2018 19:31:27 GMT] /app/src/index.ts:11:14 [DEBUG] "Many async actions" successful in 18.0s
}).catch(() => {
  // Possibility to override original log level here (maybe in success too)?
  requestSpan.fail("fatal");
  // => [Mon, 21 May 2018 19:31:27 GMT] /app/src/index.ts:14:14 [FATAL] "Many async actions" failed in 18.0s
});
```

**Comment by jlaamanen** [comment]
Alternative API proposal:
```typescript
const span = log.span.debug("Span name");
span.debug.splitTime("First split time");
span.debug.splitTime("Second split time");
span.info.success();
span.error.fail("Custom fail message");
```

This would probably be more consistent: every function would only accept an optional message. As a downside there is a bit too much repetition, as log level must be included in every call.

Another option would be to allow only one log level for each created span, and encourage users to use the normal log functions for logging actual errors/higher level logs. Example:
```typescript
const span = log.span.debug("Span name");
span.splitTime("First split time");
span.splitTime("Second split time");
span.success();
span.fail("Custom fail message");
```

As a downside, if the span was instantiated with e.g. debug level, you could not log the success time on a higher log level, which might be useful.

A combination of the two would be to have a default log level when the log span is instantiated, and add the option to override the log level if wanted:
```ts
const span = log.span.debug("Span name");
span.splitTime("First split time");
span.splitTime("Second split time");
span.info.success();
span.error.fail("Custom fail message");
```

This would limit the options of possible log level names even further (i.e. splitTime, success and fail wouldn't be available), but might be well worth the compromise.

**Comment by jlaamanen** [comment]
Implementation merged in PR #2
